Client/Wallet/SolanaProviderHelper.swift

```diff
@@ -0,0 +1,336 @@
+          let provider = tab.walletSolProvider else {
+      return (nil, buildErrorJson(status: .invalidParams, errorMessage: "Invalid args"))
+    }
+    if method == Keys.signMessage.rawValue,
+       let blobMsg = argDict[Keys.params.rawValue]?.dictionaryValue?[Keys.message.rawValue]?.numberArray {
+      /* Convert from [UInt8] to data / blob (Mojo binaryValue). */
+      var updatedParamsDict = argDict[Keys.params.rawValue]?.dictionaryValue ?? [:]
+      updatedParamsDict[Keys.message.rawValue] = MojoBase.Value(binaryValue: blobMsg)
+      argDict[Keys.params.rawValue] = MojoBase.Value(dictionaryValue: updatedParamsDict)
+    }
+    let (status, errorMessage, result) = await provider.request(argDict)
+    guard status == .success else {
+      return (nil, buildErrorJson(status: status, errorMessage: errorMessage))
+    }
+    if method == Keys.connect.rawValue,
+       let publicKey = result[Keys.publicKey.rawValue]?.stringValue {
+      // need to inject `_brave_solana.createPublickey` function before replying w/ success.
+      await tab.userScriptManager?.injectSolanaInternalScript()
+      await tab.updateSolanaProperties()
+      return (publicKey, nil)
+    } else {
+      guard let encodedResult = MojoBase.Value(dictionaryValue: result).jsonObject else {
+        return (nil, buildErrorJson(status: .internalError, errorMessage: "Internal error"))
+      }
+      return (encodedResult, nil)
+    }
+  }
+  
+  /// Given args `{serializedMessage: Buffer, signatures: {publicKey: String, signature: Buffer}}`,
+  /// will encoded the response as a json object for success or provide an error dictionary for failures
+  @MainActor func signTransaction(args: String?) async -> (Any?, String?) {
+    guard let args = args,
+          let arguments = MojoBase.Value(jsonString: args)?.dictionaryValue,
+          let serializedMessage = arguments[Keys.serializedMessage.rawValue],
+          let signatures = arguments[Keys.signatures.rawValue],
+          let provider = tab?.walletSolProvider else {
+      return (nil, buildErrorJson(status: .invalidParams, errorMessage: "Invalid args"))
+    }
+    let param = createSignTransactionParam(serializedMessage: serializedMessage, signatures: signatures)
+    let (status, errorMessage, serializedTx) = await provider.signTransaction(param)
+    guard status == .success else {
+      return (nil, buildErrorJson(status: status, errorMessage: errorMessage))
+    }
+    guard let encodedSerializedTx = JSONSerialization.jsObject(withNative: serializedTx) else {
+      return (nil, buildErrorJson(status: .internalError, errorMessage: "Internal error"))
+    }
+    return (encodedSerializedTx, nil)
+  }
+  
+  /// Given args `[{serializedMessage: Buffer, signatures: {publicKey: String, signature: Buffer}}]`,
+  /// will encoded the response as a json object for success or provide an error dictionary for failures
+  @MainActor func signAllTransactions(args: String?) async -> (Any?, String?) {
```

Comment:
Is this a standard function that we use or an additional UX button we've added for this? I'd be concerned that a secondary transaction that's not been reviewed by the user gets injected to get a faked transaction.

@kdenhartog todo: when reviewing the UX verify what this function is being used for. (I see this is something exposed by the provider, so probably worth me double checking the solana docs here too)

Reply:
Currently these are passed into the Solana provider in core and handled with the other pending requests in [CryptoStore](https://github.com/brave/brave-ios/blob/solana-dapps/Sources/BraveWallet/Crypto/Stores/CryptoStore.swift#L287-L288), which are shown as modals over the wallet panel or main wallet using [RequestContainerView](https://github.com/brave/brave-ios/blob/solana-dapps/Sources/BraveWallet/Panels/RequestContainerView.swift#L86). 

The UI isn't implemented yet for this request on iOS (Ticket for this UI [#6005](https://github.com/brave/brave-ios/issues/6005)), but the multiple transactions to be signed they are shown in a single panel within a list currently on desktop which UI we try to align with. ex. from the [test solana dapp site](https://github.com/darkdh/solana-provider-test) `Sign All Transactions (multiple)` button:
<img width="313" alt="signAllTransactions" src="https://user-images.githubusercontent.com/5314553/192795381-7a52ea04-b095-4082-a7d0-bc346c0958e5.png">
Details:
```
System Program - Transfer
From Address:
Solana Account 1
To Address:
Solana Account 1
Amount:
0.0000001 SOL

System Program - Transfer
From Address:
Solana Account 1
To Address:
Solana Account 1
Amount:
0.0000001 SOL
```

Reply:
Thanks for the comment, I got some time to look into this a bit more and realized this is more a design flaw of the API IMO than something specific to the way we've implemented it.

This seems like something where the UX could be improved here to highlight the separate transactions better, but AFAICT there's nothing specifically that's a problem here because we're still displaying all the relevant information. My biggest worry would be that an additional transaction gets injected here during the request, but as long as the wallet hasn't been compromised then the user would still be able to detect this additional transaction.

I think this is fine from my perspective, so going to resolve it.